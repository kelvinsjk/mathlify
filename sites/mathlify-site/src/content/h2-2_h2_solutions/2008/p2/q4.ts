import { Answer } from '$lib/classes/answer';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { Polynomial, sum } from 'mathlify';
import { quadraticInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { inverseRelationshipSolver } from '$content/h2-1_h2_learn/01_functions/02_inverse/05_relationship/02_practice-1';
import svg from '$static/images/h2/fns/2008p2q4a.svg?raw';
import svg2 from '$static/images/h2/fns/2008p2q4c.svg?raw';

export const answer = new Answer();

// ! Complete
// (x-4)^2 + 1

const [a, b, x] = [-4, 1, 4];
const state: Parameters<typeof quadraticInverse>[0] &
	Parameters<typeof inverseRelationshipSolver>[0] = {
	fnType: 'quadratic',
	a,
	b,
	c: 0,
	restriction: {
		type: 'right',
		x,
		inclusive: false
	},
	definition: false,
	unknownConstants: false
};
const f = sum([new Polynomial([1, a]), '^', 2], b);

// a
{
	const soln = `\`\`\` =html\n${svg}\n\`\`\`\n\n`;
	const ans = mathlifier`[Sketch](#soln-1).`;
	answer.addPart(ans, soln);
}
// b
{
	const { ans, soln } = quadraticInverse(state, f);
	answer.addPart(ans, soln);
}
// c
{
	const soln = `\`\`\` =html\n${svg2}\n\`\`\`\n\n`;
	const ans = mathlifier`[Sketch](#soln-3).`;
	answer.addPart(ans, soln);
}
// d
{
	const { ans, soln } = inverseRelationshipSolver(state, f, { lineQED: true });
	//const soln = '';
	const ansLine = mathlifier`Line ${'y=x'}.`;
	answer.addPart(ansLine + '\n\n' + ans, soln);
}
