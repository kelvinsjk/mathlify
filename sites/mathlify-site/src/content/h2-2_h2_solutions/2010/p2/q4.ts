import { Answer } from '$lib/classes/answer';
import { QED } from '$lib/typesetting/utils';
import { Expression, Polynomial } from 'mathlify';
import { mathlifierDj as mathlifier } from 'mathlifier';

import { generateFn } from '$content/h2-1_h2_learn/01_functions/01_concepts/02_domain-and-range/02_practice';
import { compositeFormula } from '$content/h2-1_h2_learn/01_functions/03_composite/02_formula/02_practice';
import svg from '$static/images/h2/fns/2010p2q4a.svg?raw';
import svg2 from '$static/images/h2/fns/2010p2q4d.svg?raw';
import { Interval, intervalBuilder } from '$content/h2-1_h2_learn/01_functions/_intervals';

export const answer = new Answer();

// TODO: answer (iv) using inequalities
// f(x) = 1/(x^2-1)
// g(x) = 1/(x-3)

const [a, b] = [1, 1];
const fState: Parameters<typeof generateFn>[0] = {
	fnType: 'special',
	a,
	b,
	c: 1,
	restriction: false,
	unknownConstants: false
	//definition: false
};
const f = generateFn(fState)[1];
const g = new Polynomial([1, -3]).reciprocal();

// a
{
	const soln = svg;
	const ans = mathlifier`[Sketch](#soln-1).`;
	answer.addPart(ans, soln);
}
// b
{
	const ans = mathlifier`Least value of ${'{k=0.}'}`;
	const soln = mathlifier`Least value of ${{}} {k = 0 ${QED}}

This is because, when ${'{k=0,}'}
the domain of ${'f'}
is ${{}} [0, 1) \\cup (1, \\infty).
\\
Under this domain, all horizontal lines ${{}}{y=k, k \\in \\mathbb{R},}
cuts the graph of ${'{y=f(x)}'}
at most once so ${'f'}
is a one-to-one function and the function ${'f^{-1}'}
exists.`;
	answer.addPart(ans, soln);
}
// c
let fg: Expression;
{
	const composite = compositeFormula([f, [], g, []], true, false, {
		noDomain: true,
		factorizeDenominator: true
	});
	const { ans, soln } = composite;
	({ exp: fg } = composite);
	answer.addPart(ans, soln);
}
// d
{
	const soln =
		mathlifier`$${fg} > 0` + svg2 + mathlifier`$${{}} 2 < x < 3 \\; \\text{ or } \\; 3 < x < 4`;
	const ans = mathlifier`${{}}{2 < x < 3}
or ${{}} {3 < x < 4.}`;
	answer.addPart(ans, soln);
}
// e
{
	const Rfg = [intervalBuilder('left', -1, false), intervalBuilder('right', 0, false)].join(
		' \\cup '
	);
	const ans = mathlifier`${'R_{fg}'} = ${Rfg}.`;
	const soln = mathlifier`$${'align*'}
	D_{g} &= ${[intervalBuilder('left', 2, false), new Interval({ left: 2, right: 3 }), new Interval({ left: 3, right: 4 }), intervalBuilder('right', 4, false)].join(' \\cup ')} \\\\
	&\\downarrow g \\\\
	R_{g} &= ${[intervalBuilder('left', -1, false), new Interval({ left: -1, right: 0 }), new Interval({ left: 0, right: 1 }), intervalBuilder('right', 1, false)].join(' \\cup ')} \\\\
	&\\downarrow f \\\\
	R_{fg} &= ${Rfg}
`;
	answer.addPart(ans, soln);
}
