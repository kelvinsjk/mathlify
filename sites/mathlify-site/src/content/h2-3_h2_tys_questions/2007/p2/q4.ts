/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Expression, pi } from 'mathlify';
import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { definiteIntegralString, xAxis } from '$lib/utils/typesetting/math';

export const question = new Question();

const upper = new Expression([[5, '/', 3], pi]);
const integral1 = definiteIntegralString('\\sin^2 x', 0, upper);
const integral2 = definiteIntegralString('\\cos^2 x', 0, upper);
const curve = `{y=x^2 \\sin x,}`;
const halfPi = new Expression([[1, '/', 2], pi]);
const line = `{x = ${halfPi}}`;

// a
question.addPart(
	mathlifier`Find the exact value of $${{}} \\displaystyle ${integral1}.

	Hence find the exact value of ${{}} \\displaystyle ${integral2}.`,
	{ marks: 6 }
);
// b
question.addPart(
	mathlifier`The region ${'R'}
is bounded by the curve ${curve}
the line ${line}
and part of the ${xAxis}
between ${'0'}
and ${halfPi}.
Find`
);
// bi
question.addSubPart(mathlifier`the exact area of ${'R'},`, { marks: 5 });
// bii
question.addSubPart(
	mathlifier`the numerical value of
the volume of revolution formed when ${'R'}
is rotated completely about the ${xAxis},
giving your answer correct to ${3}
decimal places.`,
	{ marks: 2 }
);
