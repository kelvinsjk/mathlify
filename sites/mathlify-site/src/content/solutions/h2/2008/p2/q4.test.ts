import { expect, it } from 'vitest';
import { answer } from './q4.soln';

it('2008p2q4', () => {
	expect(answer).toMatchSnapshot();
});
