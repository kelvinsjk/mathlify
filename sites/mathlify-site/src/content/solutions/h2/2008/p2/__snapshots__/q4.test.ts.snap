// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2008p2q4 1`] = `
Answer {
  "answer": {
    "parts": [
      {
        "body": "<p><a href="#soln-1">Sketch</a>.</p>
",
      },
      {
        "body": "<p><math><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><msqrt><mrow><mi>x</mi><mo>−</mo><mn>1</mn></mrow></msqrt><mo>+</mo></mrow><mrow><mn>4</mn><mi>.</mi></mrow></math>
<br>
<math><mrow><msub><mi>D</mi><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></msub><mo>=</mo></mrow><mrow><msub><mi>R</mi><mi>f</mi></msub><mo>=</mo></mrow><mrow><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mi>.</mi></mrow></math></p>
",
      },
      {
        "body": "<p><a href="#soln-3">Sketch</a>.</p>
",
      },
      {
        "body": "<p>Line <math><mrow><mi>y</mi><mo>=</mo></mrow><mrow><mi>x</mi><mi>.</mi></mrow></math></p>
<p><math><mrow><mi>x</mi><mo>=</mo></mrow><mrow><mfrac><mrow><mn>9</mn><mo>+</mo><msqrt><mn>13</mn></msqrt></mrow><mn>2</mn></mfrac><mi>.</mi></mrow></math></p>
",
      },
    ],
  },
  "solution": {
    "parts": [
      {
        "body": "<svg
	xmlns="http://www.w3.org/2000/svg"
	viewBox="-55 -60 110.00000000000001 110.00000000000001"
	class="graph svelte-1vox4to"
>
	<style>
		foreignObject {
			color: black;
			font-size: 7px;
			overflow: visible;
			white-space: nowrap;
			display: grid;
		}
		foreignObject math {
			flex-wrap: nowrap;
			transform: translate(-50%, -50%);
		}
	</style>
	<defs>
		<marker id="closed" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle cx="1" cy="1" r="1" fill="black"></circle>
		</marker>
		<marker id="open" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle
				cx="1"
				cy="1"
				r="0.75"
				fill="white"
				markerUnits="strokeWidth"
				style="stroke: black; stroke-width: 0.5"
			></circle>
		</marker>
		<marker id="arrow" markerWidth="5" markerHeight="5" refX="0" refY="1" orient="auto">
			<path d="M 0 0 L 2 1 L 0 2 z" fill="black"></path>
		</marker>
	</defs>
	<line
		x1="-50"
		x2="50"
		y1="37.654320987654316"
		y2="37.654320987654316"
		style="stroke: black; stroke-width: 2"
		marker-end="url(#arrow)"
	></line>
	<line
		x1="-37.654320987654316"
		x2="-37.654320987654316"
		y1="50"
		y2="-50"
		style="stroke: black; stroke-width: 2"
		marker-end="url(#arrow)"
	></line>
	<foreignObject x="50" y="42.654320987654316" width="1" height="1" class="svelte-1vox4to"
		><!--r4lfzw--><math xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math
		><!----></foreignObject
	><foreignObject x="-42.654320987654316" y="-52" width="1" height="1" class="svelte-1vox4to"
		><!--1ywmx8b--><math xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math
		><!----></foreignObject
	><!--[--><foreignObject
		x="-42.654320987654316"
		y="42.654320987654316"
		width="1"
		height="1"
		class="svelte-1vox4to"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<!--a38gg0--><math xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math
			><!---->
		</div></foreignObject
	><!--]--><!--[--><!--]--><!--[--><!--]--><!--[-->
	<path
		d="M11.728,25.309 12.771,25.221 13.814,24.956 14.856,24.516 15.899,23.900 16.941,23.108 17.984,22.139 19.026,20.995 20.069,19.674 21.112,18.177 22.154,16.504 23.197,14.655 24.239,12.630 25.282,10.429 26.324,8.0521 27.367,5.4989 28.410,2.7695 29.452,-0.13589 30.495,-3.2174 31.537,-6.4750 32.580,-9.9087 33.622,-13.518 34.665,-17.304 35.708,-21.266 36.750,-25.404 37.793,-29.718 38.835,-34.209 39.878,-38.875 40.920,-43.717 41.963,-48.736 "
		style="stroke: blue; fill: none"
	></path>
	<!--[--><foreignObject
		class="graph-label svelte-1vox4to"
		x="27.845679012345684"
		y="6.7975277777777805"
		width="1"
		height="1"
		><!--zcn86r--><math xmlns="http://www.w3.org/1998/Math/MathML"
			><mrow><mi>y</mi><mo>=</mo></mrow
			><mrow
				><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi
				><mo form="postfix" stretchy="false">)</mo></mrow
			></math
		><!----></foreignObject
	><!--]--><!--]--><!--[--><!--]--><!--[--><!--]--><!--[-->
	<path
		d="M11.728395061728406,25.30864197530864"
		marker-start="url(#open)"
		style="stroke: black"
	></path>
	<foreignObject
		x="11.728395061728406"
		y="30.30864197530864"
		width="1"
		height="1"
		class="svelte-1vox4to"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<!--123sqnl--><math
				><mrow
					><mo fence="true" form="prefix">(</mo><mn>4</mn><mo separator="true">,</mo><mn>1</mn
					><mo fence="true" form="postfix">)</mo></mrow
				></math
			><!---->
		</div></foreignObject
	><!--]-->
</svg>
",
      },
      {
        "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mtext>Let </mtext><mi>y</mi><mo>=</mo><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>+</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>=</mo><mi>y</mi><mo>−</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mtext>Since </mtext><mi>x</mi><mo>&gt;</mo><mn>4</mn><mo separator="true">,</mo></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>−</mo><mn>4</mn><mo>=</mo><msqrt><mrow><mi>y</mi><mo>−</mo><mn>1</mn></mrow></msqrt></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><msqrt><mrow><mi>y</mi><mo>−</mo><mn>1</mn></mrow></msqrt><mo>+</mo><mn>4</mn></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><msqrt><mrow><mi>x</mi><mo>−</mo><mn>1</mn></mrow></msqrt><mo>+</mo><mn>4</mn><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>D</mi><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msub><mi>R</mi><mi>f</mi></msub></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mspace width="0.2778em"></mspace><mi>■</mi></mrow></mtd></mtr></mtable></math></p>
",
      },
      {
        "body": "<svg
	xmlns="http://www.w3.org/2000/svg"
	class="graph svelte-1vox4to"
	viewBox="-55 -60 110.00000000000001 110.00000000000001"
>
	<style>
		foreignObject {
			color: black;
			font-size: 7px;
			overflow: visible;
			white-space: nowrap;
			display: grid;
		}
		foreignObject math {
			flex-wrap: nowrap;
			transform: translate(-50%, -50%);
		}
	</style>
	<defs>
		<marker id="closed" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle cx="1" cy="1" r="1" fill="black"></circle>
		</marker>
		<marker id="open" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle
				cx="1"
				cy="1"
				r="0.75"
				fill="white"
				markerUnits="strokeWidth"
				style="stroke: black; stroke-width: 0.5"
			></circle>
		</marker>
		<marker id="arrow" markerWidth="5" markerHeight="5" refX="0" refY="1" orient="auto">
			<path d="M 0 0 L 2 1 L 0 2 z" fill="black"></path>
		</marker>
	</defs>
	<line
		marker-end="url(#arrow)"
		x1="-50"
		x2="50"
		y1="37.654320987654316"
		y2="37.654320987654316"
		style="stroke: black; stroke-width: 2"
	></line>
	<line
		marker-end="url(#arrow)"
		x1="-37.654320987654316"
		x2="-37.654320987654316"
		y1="50"
		y2="-50"
		style="stroke: black; stroke-width: 2"
	></line>
	<foreignObject width="1" height="1" class="svelte-1vox4to" x="50" y="42.654320987654316"
		><math xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math
		><!----></foreignObject
	><foreignObject width="1" height="1" class="svelte-1vox4to" x="-42.654320987654316" y="-52"
		><math xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math
		><!----></foreignObject
	><foreignObject
		width="1"
		height="1"
		class="svelte-1vox4to"
		x="-42.654320987654316"
		y="42.654320987654316"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math
			><!---->
		</div></foreignObject
	><!----><!----><!---->
	<line
		stroke-dasharray="4"
		x1="-40.123456790123456"
		x2="48.765432098765444"
		y1="40.123456790123456"
		y2="-48.765432098765444"
		style="stroke: black"
	></line>
	<foreignObject
		class="graph-label svelte-1vox4to"
		width="1"
		height="1"
		x="6.320987654320994"
		y="-3.3209876543209944"
		><math xmlns="http://www.w3.org/1998/Math/MathML"
			><mrow><mi>y</mi><mo>=</mo></mrow
			><mrow><mi>x</mi></mrow></math
		><!----></foreignObject
	><!----><!----><!---->
	<path
		d="M11.728,25.309 12.771,25.221 13.814,24.956 14.856,24.516 15.899,23.900 16.941,23.108 17.984,22.139 19.026,20.995 20.069,19.674 21.112,18.177 22.154,16.504 23.197,14.655 24.239,12.630 25.282,10.429 26.324,8.0521 27.367,5.4989 28.410,2.7695 29.452,-0.13589 30.495,-3.2174 31.537,-6.4750 32.580,-9.9087 33.622,-13.518 34.665,-17.304 35.708,-21.266 36.750,-25.404 37.793,-29.718 38.835,-34.209 39.878,-38.875 40.920,-43.717 41.963,-48.736 "
		style="stroke: blue; fill: none"
	></path>
	<foreignObject
		class="graph-label svelte-1vox4to"
		width="1"
		height="1"
		x="28.845679012345684"
		y="6.7975277777777805"
		><math xmlns="http://www.w3.org/1998/Math/MathML"
			><mrow><mi>y</mi><mo>=</mo></mrow
			><mrow
				><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi
				><mo form="postfix" stretchy="false">)</mo></mrow
			></math
		><!----></foreignObject
	><!---->
	<path
		d="M-25.309,-11.728 -25.233,-12.694 -25.157,-13.095 -25.082,-13.402 -25.006,-13.660 -24.931,-13.888 -24.855,-14.095 -24.780,-14.284 -24.704,-14.461 -24.628,-14.626 -24.553,-14.783 -24.477,-14.932 -24.402,-15.075 -24.326,-15.211 -24.250,-15.343 -24.175,-15.470 -24.099,-15.592 -24.024,-15.711 -23.948,-15.827 -23.873,-15.939 -23.797,-16.048 -23.721,-16.155 -23.646,-16.259 -23.570,-16.361 -23.495,-16.461 -23.419,-16.558 -23.343,-16.654 -23.268,-16.748 -23.192,-16.840 -23.117,-16.930 -23.041,-17.019 -22.965,-17.107 -22.890,-17.193 -22.814,-17.278 -22.739,-17.361 -22.663,-17.443 -22.588,-17.524 -22.512,-17.604 -22.436,-17.683 -22.361,-17.761 -22.285,-17.838 -22.210,-17.914 -22.134,-17.989 -22.058,-18.063 -21.983,-18.136 -21.907,-18.209 -21.832,-18.280 -21.756,-18.351 -21.681,-18.421 -21.605,-18.490 "
		style="stroke: green; fill: none"
	></path>
	<!---->
	<path
		d="M-21.605,-18.490 -20.169,-19.694 -18.733,-20.739 -17.297,-21.674 -15.860,-22.529 -14.424,-23.320 -12.988,-24.061 -11.552,-24.760 -10.116,-25.424 -8.6798,-26.057 -7.2436,-26.662 -5.8075,-27.245 -4.3714,-27.806 -2.9352,-28.348 -1.4991,-28.873 -0.062988,-29.383 1.3731,-29.878 2.8093,-30.360 4.2454,-30.830 5.6815,-31.288 7.1177,-31.737 8.5538,-32.175 9.9899,-32.604 11.426,-33.024 12.862,-33.437 14.298,-33.841 15.734,-34.238 17.171,-34.629 18.607,-35.013 20.043,-35.390 21.479,-35.762 22.915,-36.128 24.351,-36.489 25.787,-36.844 27.223,-37.195 28.660,-37.541 30.096,-37.882 31.532,-38.219 32.968,-38.551 34.404,-38.880 35.840,-39.204 37.276,-39.525 38.713,-39.842 40.149,-40.156 41.585,-40.466 43.021,-40.773 44.457,-41.076 45.893,-41.377 47.329,-41.674 48.765,-41.969 "
		style="stroke: green; fill: none"
	></path>
	<foreignObject
		class="graph-label svelte-1vox4to"
		width="1"
		height="1"
		x="-21.60493827160494"
		y="-38.49040194450822"
		><math xmlns="http://www.w3.org/1998/Math/MathML"
			><mrow><mi>y</mi><mo>=</mo></mrow
			><mrow
				><msup
					><mi>f</mi
					><mrow
						><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow
					></msup
				><mo form="prefix" stretchy="false">(</mo><mi>x</mi
				><mo form="postfix" stretchy="false">)</mo></mrow
			></math
		><!----></foreignObject
	><!----><!----><!----><!---->
	<path
		marker-start="url(#open)"
		style="stroke: black"
		d="M11.728395061728406,25.30864197530864"
	></path>
	<foreignObject
		width="1"
		height="1"
		class="svelte-1vox4to"
		x="11.728395061728406"
		y="30.30864197530864"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<math
				><mrow
					><mo fence="true" form="prefix">(</mo><mn>4</mn><mo separator="true">,</mo><mn>1</mn
					><mo fence="true" form="postfix">)</mo></mrow
				></math
			><!---->
		</div></foreignObject
	>
	<path
		marker-start="url(#open)"
		style="stroke: black"
		d="M-25.30864197530864,-11.728395061728406"
	></path>
	<foreignObject
		width="1"
		height="1"
		class="svelte-1vox4to"
		x="-25.30864197530864"
		y="-6.728395061728406"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<math
				><mrow
					><mo fence="true" form="prefix">(</mo><mn>1</mn><mo separator="true">,</mo><mn>4</mn
					><mo fence="true" form="postfix">)</mo></mrow
				></math
			><!---->
		</div></foreignObject
	><!---->
</svg>
",
      },
      {
        "body": "<p>The graph of <math><mrow><mi>y</mi><mo>=</mo><msup><mi>f</mi><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false" lspace="0em" rspace="0em">)</mo></mrow></math>
is a reflection of the graph of <math><mrow><mi>y</mi><mo>=</mo><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false" lspace="0em" rspace="0em">)</mo></mrow></math>
about the line <math><mrow><mrow><mi>y</mi><mo>=</mo><mi>x</mi></mrow><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
<p>We observe that the intersection
between the two curves is
also the intersection between the graph of
<math><mrow><mi>y</mi><mo>=</mo><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false" lspace="0em" rspace="0em">)</mo></mrow></math>
and the line <math><mrow><mrow><mi>y</mi><mo>=</mo><mi>x</mi></mrow><mi>.</mi></mrow></math></p>
<p>Hence the solution to <math><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><msup><mi>f</mi><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false" lspace="0em" rspace="0em">)</mo></mrow></math>
can be found by solving</p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>+</mo><mn>1</mn><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>x</mi><mn>2</mn></msup><mo>−</mo><mn>8</mn><mi>x</mi><mo>+</mo><mn>17</mn><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>x</mi><mn>2</mn></msup><mo>−</mo><mn>9</mn><mi>x</mi><mo>+</mo><mn>17</mn><mo>=</mo><mn>0</mn></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mi>x</mi></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mn>9</mn><mo fence="true" form="postfix">)</mo></mrow><mo>±</mo><msqrt><mrow><msup><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mn>9</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>−</mo><mn>4</mn><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mrow><mo fence="true" form="prefix">(</mo><mn>17</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></msqrt></mrow><mrow><mn>2</mn><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mn>9</mn><mo>±</mo><msqrt><mn>13</mn></msqrt></mrow><mn>2</mn></mfrac></mrow></mtd></mtr></mtable></math></p>
<p>Since <math><mrow><mi>x</mi><mo>&gt;</mo></mrow><mrow><mn>4</mn><mo separator="true">,</mo></mrow></math>
<math display="block" class="tml-display" style="display:block math;"><mrow><mi>x</mi><mo>=</mo><mfrac><mrow><mn>9</mn><mo>+</mo><msqrt><mn>13</mn></msqrt></mrow><mn>2</mn></mfrac><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
",
      },
    ],
  },
}
`;
